src/rqt_gauges_2/speedometer_widget.py

```diff
@@ -0,0 +1,133 @@
+    :param slot_num: index of slot to return, ``str``
+    :returns: fn(msg_field)->msg_field[slot_num]
+    """
+    def fn(f):
+        return getattr(f, field_name).__getitem__(slot_num)
+    return fn
+
+def _field_eval(field_name):
+    """
+    :param field_name: name of field to return, ``str``
+    :returns: fn(msg_field)->msg_field.field_name
+    """
+    def fn(f):
+        return getattr(f, field_name)
+    return fn
+
+
+def generate_field_evals(fields):
+    evals = []
+    fields = [f for f in fields.split('/') if f]
+    for f in fields:
+        if '[' in f:
+            field_name, rest = f.split('[')
+            slot_num = int(rest[:rest.find(']')])
+            evals.append(_array_eval(field_name, slot_num))
+        else:
+            evals.append(_field_eval(f))
+    return evals
+
+class SpeedometerWidget(QWidget):
+    def __init__(self, node, initial_topics=None):
+        super(SpeedometerWidget, self).__init__()
+        self.setObjectName('Speedometer_widget')
+
+        self.node = node
+        self.sub = None
+
+        _, package_path = get_resource('packages', 'rqt_gauges_2')
+        ui_file = os.path.join(package_path, 'share', 'rqt_gauges_2', 'resource', 'speedometer.ui')
+        loadUi(ui_file, self)
+
+        self._topic_completer = TopicCompleter(self.topic_to_subscribe)
+        self._topic_completer.update_topics(node)
+        self.topic_to_subscribe.setCompleter(self._topic_completer)
+
+        # Objects Properties
+        self.max_value.setAlignment(Qt.AlignCenter)
+        self.min_value.setAlignment(Qt.AlignCenter)
+
+        self.max_value.setPlaceholderText(str(self.speedometer_gauge.maxValue))
+        self.min_value.setPlaceholderText(str(self.speedometer_gauge.minValue))
+
+        # Signals Connection
+        self.min_value.textChanged.connect(self.updateMinValue)
+        self.max_value.textChanged.connect(self.updateMaxValue)
+        self.units.currentTextChanged.connect(self.updateUnits)
+        self.subscribe_button.pressed.connect(self.updateSubscription)
+
+    @pyqtSlot()
+    def updateMinValue(self):
+        new_min_value = self.min_value.toPlainText()
+        if new_min_value.isnumeric():
+            self.speedometer_gauge.setMinValue(int(new_min_value))
+        else:
+            self.speedometer_gauge.setMinValue(0)
+
+    @pyqtSlot()
+    def updateMaxValue(self):
+        new_max_value = self.max_value.toPlainText()
+        if new_max_value.isnumeric():
+            self.speedometer_gauge.setMaxValue(int(new_max_value))
+        else:
+            self.speedometer_gauge.setMaxValue(180)
+
+    @pyqtSlot(str)
+    def updateUnits(self, new_units):
+        self.speedometer_gauge.units = new_units
+        self.speedometer_gauge.update()
+
+    @pyqtSlot()
+    def updateSubscription(self):
+        if self.node.destroy_subscription(self.sub):
+            print("Previous subscription deleted")
+        else:
+            print("There was no previous subscription")
```

Comment:
The other rqt plugins, what kinds of loggers are they using for these kinds of prints?

Reply:
From checking [RQT_Plot](https://github.com/ros-visualization/rqt_plot/tree/foxy-devel/src/rqt_plot) I only noticed some prints the same way I'm doing it. However, I could get the node logger and log the same information.